circle.yml

```diff
@@ -0,0 +1,16 @@
+dependencies:
+  cache_directories:
+    - /usr/local/Homebrew
+  override:
+    - brew install shellcheck
+
+machine:
```

Comment:
Cool, cool. I had seen Docker as the headline feature but agree maybe better to merge this so @derekprior can use it for his active changes on other PRs and keep our eye on Circle 2.0 as we (I?) get more experience with it.